### ISSUE TITLE
Data display issues 

### ISSUE THREAD

**Issue by nekromant** [alexstocker/sensorlogger#39]
Hi, thanks for your effort to create such a nice app. 

I'm currently writing an esp8266 library for interfacing with sensorlogger (I plan to opensource once it's ready, part of my [frankenstein firmware for esp8266](https://github.com/nekromant/esp8266-frankenstein)), and have run in a quite an issue. I'm able to register the device and post the data, and can even see all the data appearing in sensorlogger_logs database. But the UI is somehow messed up, as if the deviceDataTypes portion of 'register device' api request was not handled properly: e.g. in the chart for the device it shows humidity and temperature, although no such data types exist for this device. Same goes for the list of data values.

(The current device is right now just a dumb mockup, posting random numbers).  

## Expected Behavior

GUI should not show data registered for the device, same goes for data view. 

## Actual Behavior

GUI shows only temperature and humidity for every device.

## Steps to Reproduce the Problem

Can be done by posting the following JSON to the server : 
```
frankenstein > svclog register                                                                                                                                            
Registering device at https://cloud.ncrmnt.org/index.php/apps/sensorlogger/api/v1/registerdevice/                                                                         
{                                                                                                                                                                         
        "deviceId":     "dev-arven-001",                                                                                                                                  
        "deviceName":   "arven-green",                                                                                                                                    
        "deviceType":   "dummy",                                                                                                                                          
        "deviceGroup":  "development",                                                                                                                                    
        "deviceParentGroup":    "",                                                                                                                                       
        "deviceDataTypes":      {                                                                                                                                         
                "type": "mana",                                                                                                                                           
                "description":  "Mana regen rate",                                                                                                                        
                "unit": "pts/sec"                                                                                                                                         
        }                                                                                                                                                                 
}

frankenstein > svclog post                                                                                                                                                
Posting data to https://cloud.ncrmnt.org/index.php/apps/sensorlogger/api/v1/createlog/                                                                                    
{                                                                                                                                                                         
        "deviceId":     "dev-arven-001",                                                                                                                                  
        "mana": "39.52"                                                                                                                                                   
}
```
## Specifications

  - Version: 0.0.5 (from nextcloud appstore)
  - Platform: nextcloud 12.0.4
  - PHP 7.0.19-1 (debian stretch)

P.S. I've also noticed that if I omit deviceParentGroup from json query I'll get an exception with data partially(?) added to the db.

**Comment by nekromant** [comment]
Thanks for the quick response, going to try right now. A quick question: Should 
Extended createLog always supply values for all available dataTypes or I can send it in chunks? (e.g. a createLog with only dataType 2, followed by a request with only dataType 3)

**Comment by alexstocker** [comment]
Yes you can send chunks but jqPlot can not draw the resulting chart correctly at the moment.

**Comment by alexstocker** [comment]
A possible solution or workaround could be splitting up sensor to kind of "virtual" sensors

* For Instance:
   *  Register dev-arven-001-01 for CO2, NOX
   *  Register dev-arven-001-02 for CO, H

**Comment by nekromant** [comment]
Again thanks a lot for the help, I reworked my API so that it would deliver all available data in one go. More memory hungry, since I have to buffer the full json in ram, but I guess esp8266 will endure. 

It kind of works now, though (if I got it right) I still have to issue a request go get all the correct dataTypeId's from the server. Suggestion: Using relative ids (e.g. the order of elements in the array when registering the device, will make the API more friendly towards small IOT devices.
